geth/testing/testing.go

```diff
@@ -91,15 +105,32 @@ func (s *BaseTestSuite) StartTestNode(networkID int, opts ...TestNodeOption) {
 	require.True(s.NodeManager.IsNodeRunning())
 }
 
-// TestNodeOption is a callback passed to StartTestNode which alters its config.
-type TestNodeOption func(config *params.NodeConfig)
+// EnsureNodeSync ensures that synchronization of the node is done once and that it
+// is done properly else, the call will fail.
+func (s *BaseTestSuite) EnsureNodeSync(forceResync ...bool) {
+	if len(forceResync) > 0 && forceResync[0] {
+		s.nodeSyncCompleted = false
+	}
 
-// WithUpstream returns TestNodeOption which enabled UpstreamConfig.
-func WithUpstream(url string) TestNodeOption {
-	return func(config *params.NodeConfig) {
-		config.UpstreamConfig.Enabled = true
-		config.UpstreamConfig.URL = url
+	if s.nodeSyncCompleted {
+		return
 	}
+
+	require := s.Require()
+
+	ethClient, err := s.NodeManager.LightEthereumService()
+	require.NoError(err)
+	require.NotNil(ethClient)
+
+	sync := node.NewSyncPoll(ethClient)
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Hour)
```

Comment:
Let's say `15 * time.Minute`. Test timeout is 40m in our case. If synchronisation is taking longer than 15 minutes, we should, probably, fix that by updating CHT or something.